### PR TITLE
CU-86azwdqc2 - Store store_id used when syncing shipment and always u…

### PR DESCRIPTION
**Pull request by mrshahsuvie**
…se that store_id for all subsequent syncs

### CODE REVIEW

**Comment by NewAlexandria** [comment]
Task linked: [CU-86azwdqc2 Store store_id used when syncing shipment and always use that store_id for all subsequent syncs](https://app.clickup.com/t/86azwdqc2)

File: lib/solidus_shipstation/api/batch_syncer.rb [hunk 1]
```diff
@@ -54,6 +54,7 @@ def post_sync(shipstation_order, shipments)
         shipment.update_columns(
           shipstation_synced_at: Time.zone.now,
           shipstation_order_id: shipstation_order['orderId'],
+          store_id: shipstation_order['advancedOptions']['storeId']
```
Comment (NewAlexandria) [inline lib/solidus_shipstation/api/batch_syncer.rb]: What if it's null / not passed?

File: lib/solidus_shipstation/api/batch_syncer.rb [hunk 2]
```diff
@@ -54,6 +54,7 @@ def post_sync(shipstation_order, shipments)
...
         )
```
Comment (NewAlexandria) [inline lib/solidus_shipstation/api/batch_syncer.rb]: I think you need to include this only if it's passed.

```suggestion
        sync_opts = {
          shipstation_synced_at: Time.zone.now,
          shipstation_order_id: shipstation_order['orderId']
        }
        store_id = shipstation_order.dig('advancedOptions', 'storeId')
        sync_opt[:store_id] = store_id if store_id
        shipment.update_columns(sync_opt)
```

File: lib/solidus_shipstation/api/batch_syncer.rb (hunk 1 above)
Comment (mrshahsuvie) [reply lib/solidus_shipstation/api/batch_syncer.rb]: I think we always get the store ID from `OrderSelectorSerializer`, but it wouldn't hurt to add a check

**Review by NewAlexandria — APPROVED** [review]

**Pull request merged**